backend/handlers/upload.py

```
@@ -242,29 +242,30 @@ def _vector_ds(fh, ds_name, ds_kind, meta_kwargs, meta_pkeys, resample,
     spectra = spectra[:, mask]
 
   if ds_kind == 'LIBS' and wave.shape[0] not in (6144, 6143, 5485):
-    return (415, 'Wrong number of channels for LIBS data: %d.' % wave.shape[0])
+    return 415, 'Wrong number of channels for LIBS data: %d.' % wave.shape[0]
 
   # make sure there's no whitespace sticking to the pkeys
   pkey = np.char.strip(pkey)
 
+  # Check length and see if the original arrays are the same
   if len(meta_pkeys) > 0 and not np.array_equal(meta_pkeys, pkey):
     if len(meta_pkeys) != len(pkey):
-      return (415, 'Spectrum and metadata names mismatch.',
-                   'wrong number of meta_pkeys for vector data')
+      return 415, 'Spectrum and metadata names mismatch.', 'wrong number of meta_pkeys for vector data'
+
     meta_order = np.argsort(meta_pkeys)
     data_order = np.argsort(pkey)
     if not np.array_equal(meta_pkeys[meta_order], pkey[data_order]):
-      return (415, 'Spectrum and metadata names mismatch.')
+      return 415, 'Spectrum and metadata names mismatch.'
     # convert data to meta order
     order = np.zeros_like(data_order)
     order[data_order[meta_order]] = np.arange(len(order))
     data = data[order]
-    assert np.array_equal(meta_pkeys, pkey[order])
+    # assert np.array_equal(meta_pkeys, pkey[order]) # Causes it to silently fail in some cases
```

I think we should keep the check, but instead of using `assert` it should return a 415 error with a useful message.

If these arrays aren't equal, it means that the spectra and their associated metadata are out of sync, and your uploaded dataset isn't going to be correct.